packages/react-components/react-dialog/src/components/Dialog/useDialog.ts

```diff
@@ -54,9 +51,19 @@ export const useDialog_unstable = (props: DialogProps): DialogState => {
     requestOpenChange,
   });
 
-  const handlebackdropClick = useEventCallback((event: React.MouseEvent<HTMLDivElement>) => {
+  const { disableBodyScroll, enableBodyScroll } = useDisableBodyScroll();
+
+  React.useEffect(() => {
+    if (open) {
+      disableBodyScroll();
+    } else {
+      enableBodyScroll();
+    }
+  }, [disableBodyScroll, enableBodyScroll, open]);
```

Comment:
Did you consider API that returns a callback to call? I.e.:

```tsx
const disableBodyScroll = useDisableBodyScroll();
const enableBodyScroll = disableBodyScroll();
```

We will have a single function in return and their usage will be coupled.

```suggestion
  const disableBodyScroll = useDisableBodyScroll();

  React.useEffect(() => {
    if (open) {
      return disableBodyScroll();
    }
  }, [disableBodyScroll, open]);
```

Reply:
> Did you consider API that returns a callback to call? I.e.:

Sure, sound good. I'll take a look into that